### ISSUE TITLE
pad fails without separator when decimal places are zero

### ISSUE THREAD

**Issue by Hamper** [avoidwork/filesize.js#184]
The 'pad' method doesn't work when the separator is not set and all decimal places are zero.
```
> filesize(1100)
'1.1 kB'
> filesize(1100, {round: 2, pad: true})
'1.10 kB'
> filesize(1100, {round: 2, pad: true, separator: '.'})
'1.10 kB'

> filesize(1001)
'1 kB'
> filesize(1001, {round: 2, pad: true})
'1 kB' // must be '1.00 kB'
> filesize(1001, {round: 2, pad: true, separator: '.'})
'1.00 kB'
```

**Comment by avoidwork** [comment]
I think you want to use `precision`; the default value of `separator`  is an empty string so that it doesn't count as an incomplete float when the string concatenation happens.

The `precision` value always forces the number of digits; it's just `Number.toPrecision()` on the calculated number, and works well if you're not dealing with zetabyte values that exceed the precision.

```javascript
filesize(1001, {precision: 3}); // 1.00 kB
filesize(1024, {precision: 3, base: 2}); // 1.00 KiB
```

**Comment by avoidwork** [comment]
A `precision` of 3 will give you `xxx`, `xx.x`, `x.xx`, & `.xxx` as potential formatted outputs. I _think_ this is what most users would be after if you want always to have decimal numbers.

**Comment by Hamper** [comment]
I need to always have 2 decimal places, for example
```
  x.xx
 xx.xx
xxx.xx
```
And this work when `separator` is not empty, but don't work with default value

`precision` also has other potential problems like this:
```
> filesize(110, {precision: 2})
'1.1e+2 B'
```

**Comment by avoidwork** [comment]
This is fixed in 10.1.5 👍🏻
